src/V3Options.cpp

```diff
@@ -657,7 +657,9 @@ string V3Options::getenvPERL() {  //
 string V3Options::getenvSYSTEMC() {
     string var = V3Os::getenvStr("SYSTEMC", "");
     if (var == "" && string(DEFENV_SYSTEMC) != "") {
-        var = DEFENV_SYSTEMC;
+        // treat compiled-in DEFENV string literals as C-strings to enable
+        // binary patching for relocatable installs (e.g. conda)
+        var = string{DEFENV_SYSTEMC}.c_str();
```

Comment:
@wsnyder, when I went in to use an initialization list as you suggested, I thought it might make sense to also use it for the other object that is compared against empty string or perhaps hoist the creation of a string object for the DEFENV variable out of the conditional so that only one is created.  I went for the minimal patch first but I'd be happy to go back and do it this way instead:
```c
    // treat compiled-in DEFENV string literals as C-strings to enable
    // binary patching for relocatable installs (e.g. conda)
    string defenv = string{DEFENV_SYSTEMC}.c_str();
    if (var == "" && defenv != "") {
        var = defenv;
```

Reply:
Yes, I think it's better to do it also for the compare.   Nit: Please capitalize "treat".

Reply:
Addressed in [d92a953](https://github.com/verilator/verilator/pull/4044/commits/d92a953b8d40a3a8ef247c3a0c55d69fc5593276)